src/main/java/seedu/address/logic/commands/EditIsolatedEventCommand.java

```diff
@@ -0,0 +1,171 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ENDDATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ISOEVENT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STARTDATETIME;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.IsolatedEvent;
+import seedu.address.model.person.Person;
+
+/**
+ * Edits the details of an existing event in the person's isolated event list.
+ */
+public class EditIsolatedEventCommand extends Command {
+    public static final String COMMAND_WORD = "ie_edit";
+    public static final String MESSAGE_SUCCESS = "Isolated event edited to: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edit an isolated event into the isolated event list"
+            + "by the index number used in the last person listing.\n "
+            + "Parameters: "
+            + "[index number of the person], "
+            + "[index number of the event] in the event list"
+            + PREFIX_ISOEVENT + "EVENT_NAME"
+            + PREFIX_STARTDATETIME + "START_DATE"
+            + PREFIX_ENDDATETIME + "END_DATE"
+            + "Example: " + COMMAND_WORD + "1" + " 1" + "ie/biking";
+
+    private final Index personIndex;
+    private final Index eventIndex;
+    private final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * Constructor for EditIsolatedCommand.
+     * @param personIndex in which the event is contained in.
+     * @param eventIndex for the event to be edited.
+     * @param editEventDescriptor of what to be edited.
+     */
+    public EditIsolatedEventCommand(Index personIndex, Index eventIndex, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(personIndex);
+        requireNonNull(eventIndex);
+        requireNonNull(editEventDescriptor);
```

Comment:
Not sure if the standard is to make all these 1 line or not but I think keeping it separate would help with debugging! Nice!